### DISCUSSION TITLE
开发者例会议题收集（20220712）

### DISCUSSION THREAD

**Discussion by andrewshan** [polarismesh/polaris#496] [Meeting]
1. metadata相互覆盖的问题：https://github.com/polarismesh/polaris/issues/488
2. TLS相关需求和方案讨论
  a. 北极星支持mTLS方案讨论：https://github.com/polarismesh/polaris/issues/422
  b. discover接口支持TLS：https://github.com/polarismesh/polaris/pull/456
  c. redis支持TLS：https://github.com/polarismesh/polaris/issues/398

**Comment by Guo-Zhang** [comment]
关于量潮微服务系统的场景介绍：

我们把我们的整个软件**系统**定义为“量潮应用系统”（QtApps），计划包括“量潮课堂”、“量潮数据服务”等外部**应用**（即“前台”）、“文档平台”等内部应用（即“后台”）、“用户鉴权平台”、“支付平台”等领域层应用（即“中台”）。每个应用通常有一个或者多个**服务**，比如量潮课堂包括主站和管理后台，分别为前台和后台应用。

我们采取云原生的**“基于API的协作”**的原则设计**服务**之间的协同。常见的场景包括，前后台消费中台API（比如所有的鉴权都需要“用户鉴权平台”处理），或者后台服务消费前台服务API（比如“量潮课堂”的管理后台消费主站的API），或者前台服务互相消费（比如”量潮数据服务“打包”量潮课堂“API，业务层面实现把课程打包到项目卖）。

从DevOps的流程看，我们需要解决“设计->开发->构建->测试->发布->部署->监控->运维”的方方面面的问题。目前的流程为：
1. 设计。我们采取“API First Design”的策略，先设计API、再实现。通过Coding提供的OpenAPI支持生成API文档和Mock服务。
2. 开发。我们使用Django为主要开发框架，以Serverless为辅。
3. 测试。单元测试使用框架自己的能力，服务测试和端到端测试暂时手动为主。
4. 发布。我们以Coding提供的Git仓库的发布为主要发布，并关联流水线实现CICD。
5. CICD。我们使用微信云托管作为主要的CICD平台。
6. 监控和运维。目前使用云托管自带的基础功能，计划通过**北极星**提高相关能力。

目前的主要困难在于缺少一套贯穿设计、测试、监控等环节的**系统级**微服务治理能力。

理想的情况是：
1. API文档以团队评审的形式固定，作为指导整个DevOps周期的参考。
2. API文档可以用来直接规范测试和监控，避免重复写验证的同时提高验证质量。
3. API文档和云平台匹配，最大限度地使用云平台的能力实现“云原生"，减少自己维护的环节。

眼下的问题:
1. 缺少围绕OpenAPI做的API测试工具（类似于Apifox），端到端测试和服务测试不易编排和验证。不用Apifox的主要考虑是不好把它整合进整个体系。支持OpenAPI的CLI工具最佳，可以围绕CLI做流水线。
2. 监控和运维能力空白。主要是北极星还没有顺利地接入，需要我们先把Python SDK造出来，Sidecar接入以后依然需要Python SDK二次接入适配Django框架。
3. 云原生化以后的运维不好做了。比如Django命令行工具不可用，失去了一大套运维工具。

**Comment by andrewshan** [comment]
会议主题: Polaris社区开发者双周会
会议时间: 2022-07-12 19:30 - 20:00

点击链接入会: https://meeting.tencent.com/dm/kBemx9IzxmGE

会议 ID: 701 1815 5065

**Comment by andrewshan** [comment]
2. TLS相关需求和方案讨论
a. 北极星支持mTLS方案讨论：https://github.com/polarismesh/polaris/issues/422
b. discover接口支持TLS：https://github.com/polarismesh/polaris/pull/456
c. redis支持TLS：https://github.com/polarismesh/polaris/issues/398
===
这里面有2个层面：
1. 控制面证书签发：由polaris自己签发一个根证书，数据面拿到这个根证书，通过挂configmap以及配置的方式，挂载到数据面，数据面通过该根证书进行通信。
2. 数据面证书签发：通过polaris-security进行签发

![image](https://user-images.githubusercontent.com/45474304/178478740-f7abbb52-6b2f-40d0-8290-f7988f839b99.png)

**Reply by chuntaojun** [reply]
提问：这个数据面的证书可以自己签嘛

后续支持

**Comment by chuntaojun** [comment]
polaris-python sdk 进展

- 熟悉现有的 SDK 实现原理

**Comment by chuntaojun** [comment]
## instance metadata 修改问题

- 通过flag的标识让用户自己选择
- 属性的生命周期来进行区分
- 具体还是需要依赖具体场景，等 @shichaoyuan 后续在给出跟具体的一些场景情况


## 北极星支持mTLS方案讨论

- 遗留问题：数据面证书是否可以支持用户自己签发？

相关PR：https://github.com/polarismesh/polaris-security/pull/1

## 量潮微服务系统

- polaris-python 还在属性现有 SDK 的实现原理
- 果哥还在整理 mock 方案
